tck/src/main/java/org/hiero/tck/util/JSONRPCParamParser.java

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.tck.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import org.hiero.tck.exception.JSONRPCParseException;
+import org.hiero.tck.methods.sdk.param.CommonTransactionParams;
+import org.hiero.tck.methods.sdk.param.CustomFee;
+
+public class JSONRPCParamParser {
+
+    public static Optional<CommonTransactionParams> parseCommonTransactionParams(Map<String, Object> jrpcParams) {
+        try {
+            return parseJsonObject(jrpcParams, "commonTransactionParams", CommonTransactionParams::parse);
+        } catch (Exception e) {
+            throw new JSONRPCParseException("Failed to parse CommonTransactionParams", e);
+        }
+    }
+
+    public static Optional<List<CustomFee>> parseCustomFees(Map<String, Object> jrpcParams) {
+        try {
+            return parseJsonArray(jrpcParams, "customFees", jsonObj -> {
+                try {
+                    return new CustomFee().parse(jsonObj);
+                } catch (Exception e) {
+                    throw new JSONRPCParseException("Failed to parse CustomFee", e);
+                }
+            });
+        } catch (Exception e) {
+            throw new JSONRPCParseException("Failed to parse CustomFees array", e);
+        }
+    }
```

Comment:
I think we can just throw the exception, because the error message will get overridden anyways and we don't have specific handling for `JSONRPCParseException`
```suggestion
    public static Optional<List<CustomFee>> parseCustomFees(Map<String, Object> jrpcParams) throws Exception {
            return parseJsonArray(jrpcParams, "customFees", jsonObj -> new CustomFee().parse(jsonObj));
    }

```